controllers/argocd/service.go

```diff
@@ -461,6 +448,27 @@ func (r *ReconcileArgoCD) reconcileServerService(cr *argoproj.ArgoCD) error {
 	if err := controllerutil.SetControllerReference(cr, svc, r.Scheme); err != nil {
 		return err
 	}
+
+	existingSVC := &corev1.Service{}
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, existingSVC) {
+		if !cr.Spec.Server.IsEnabled() {
+			return r.Client.Delete(context.TODO(), svc)
+		}
+		if ensureAutoTLSAnnotation(r.Client, svc, common.ArgoCDServerTLSSecretName, cr.Spec.Server.WantsAutoTLS()) {
+			return r.Client.Update(context.TODO(), svc)
+		}
+		if !cr.Spec.Repo.IsEnabled() {
+			return nil
+		}
+		changed := false
+		if !reflect.DeepEqual(svc.Spec.Type, existingSVC.Spec.Type) {
+			changed = true
+		}
+		if changed {
+			return r.Client.Update(context.TODO(), svc)
+		}
+		return nil
+	}
```

Comment:
```suggestion
		if !reflect.DeepEqual(svc.Spec.Type, existingSVC.Spec.Type) {
		        existingSVC.Spec.Type = svc.Spec.Type
			changed = true
		}
		if changed {
			return r.Client.Update(context.TODO(), existingSVC)
		}
		return nil  // no changes detected, do nothing
	}
```
While updating, we should prefer using existing resource obj `existingSVC` in `Update` with modified fields using data from our latest constructed source of truth obj i.e `svc`